### PR TITLE
Make az.guestconfiguration cmdlet support ARC senario

### PR DESCRIPTION
**Pull request by gehuang-dut**
## Description

az.guestconfiguration currently support Azure VM senario only. This change is to make it support ARC machine senario as well.

Design sign off:
https://1drv.ms/w/s!AsGgVeIhx2Pgg6Qb99zBroo4lz0Crw?e=dWW9Hw

GuestConfiguration Service Swagger PR: (approved and merged in master)
https://github.com/Azure/azure-rest-api-specs/pull/9595/

GuestConfiguration SDK PR: (auto generated by autorest)
https://github.com/Azure/azure-sdk-for-net/pull/12373

----------
NOTE:
This change needs to upgrade guestconfiguration SDK nuget package first, which is pending publishing. I'll add this update once the new SDK published. Currently the build is meant to fail.
 

-------------

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/GuestConfiguration/GuestConfiguration/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,9 @@
 -->
 ## Upcoming Release
 
+## Version 0.10.8
```
Comment (VeryEarly) [inline src/GuestConfiguration/GuestConfiguration/ChangeLog.md:22]: no need to update version, please move message under "upcoming release" section

**Comment by VeryEarly** [comment]
Also please resolve the build issue

File: src/GuestConfiguration/GuestConfiguration/ChangeLog.md (hunk 1 above)
Comment (gehuang-dut) [reply src/GuestConfiguration/GuestConfiguration/ChangeLog.md]: updated.

**Comment by gehuang-dut** [comment]
> 
> 
> Also please resolve the build issue

re-recorded the tests. The Linux failure is caused by a dotnet issue. I think it's a test platform issue.

**Comment by VeryEarly** [comment]
> > Also please resolve the build issue
> 
> re-recorded the tests. The Linux failure is caused by a dotnet issue. I think it's a test platform issue.

It's known random failure, re-run CI should fix it.

**Comment by gehuang-dut** [comment]
> 
> 
> /azp run

Got it. Yes, passed. THanks

**Pull request merged**